.github/workflows/ci.yml

```diff
@@ -23,12 +23,6 @@ jobs:
       - name: "Install Python"
         uses: actions/setup-python@v2
 
-      - name: "Install pre-commit"
-        run: pip install pre-commit
-
-      - name: "Run pre-commit checks"
-        run: pre-commit run --all-files
-
```

Comment:
If those steps are removed, then I assume `checks` jobs is redundant. Outstanding jobs step be moved to `build` job.

Reply:
However, it would be nice to still execute pre-commit checks in `checks` job. Then `build` could be run only if `checks` job is successful. It will reduce runners usage.

Reply:
@hovaesco is it necessary to split out the checks vs the build? I agree that one could argue that `pre-commit` performs pre-flight checks, however i) it does mean the DRY principle isn't really adhered to, and ii) you could argue (per the name) that these should actually be first run locally (alongside all the other local CI checks) at the time of the commit. This is actually the most efficient use of resources and results in faster iteration.

Reply:
You cannot control behaviour of people who submit PRs. It's still useful to let people submit PRs in an imperfect shape and guide them to the finish line so that their next contribution is easier.

Also keeping the checks split has two benefits:
- Lesser wall-time for each check.
- Easy to see at a glance what failed with zero additional clicks (that's also why a matrix was used instead of tox for the version based tests)